### ISSUE TITLE
RHP3 websocket support

### ISSUE THREAD

**Issue by n8maninger** [SiaFoundation/hostd#22]
Add web socket support to RHP3 so hosts can communicate directly with browsers. This feature was started in `siad`, but had some unresolved problems with the implementation preventing actual use.

**Unresolved:**
+ How will hosts manage their SSL certificates? 
  + There is a LetsEncrypt Go library; we could have cert management built-in, like Caddy. Some hosts will want to override the automatic management.
  + Require hosts to set up a proxy, but that is much less friendly.
+ This would effectively require our hosts to announce using a domain instead of an IP. 
  + We already recommend using a domain name over an IP, but this would change that to a requirement.

**Tasks:**
+ [x] Make sure SiaFoundation/mux and SiaFoundation/core transports work over a WebSocket conn.
+ [x] Add the web socket port, docker port, and CLI flag for `9984/tcp`
+ [x] Start a secondary RHP3 listener on the WebSocket port.
+ [ ] Cert management

**Extra:**
+ [ ] Provide a browser RHP3 client lib. This may be possible with Go WASM without any changes, but I'm not sure this is the best option.

**Comment by n8maninger** [comment]
All tasks have been completed except for automatic cert management. Currently, the cert path is hardcoded to `certs/rhp3.crt` and `certs/rhp3.key`. If no cert is present, a self-signed certificate is generated.

**Comment by pcfreak30** [comment]
As I'm not sure if you guys are aware, but found this as part of my own research, https://pkg.go.dev/golang.org/x/crypto/acme/autocert.

The SSL part should not be that complicated.

Kudos.

**Comment by n8maninger** [comment]
Meant to reply, not delete…

> As I'm not sure if you guys are aware, but found this as part of my own research, https://pkg.go.dev/golang.org/x/crypto/acme/autocert.
The SSL part should not be that complicated.
Kudos.

There were a few flow annoyances I ran into with `golang.org/x/crypto/acme/autocert` that prevented us from using it directly. We also looked at `github.com/mholt/acmez` and `github.com/caddyserver/certmagic` as alternatives. We ended up rolling our own manager with `golang.org/x/crypto/acme`.
